### PR TITLE
Brute speed up

### PR DESCRIPTION
**Pull request by jakubste**

### CODE REVIEW

File: ecs/elections/algorithms/brute_force.py [hunk 1]
```diff
@@ -1,21 +1,49 @@
 from ecs.elections.algorithms.algorithm import Algorithm
+from ecs.utils.math import ell_p_norm
 
 
 class BruteForce(Algorithm):
+    preferences = None
+
+    def fetch_preferences(self):
```
Comment (jakubste) [inline ecs/elections/algorithms/brute_force.py]: test me

File: ecs/elections/algorithms/brute_force.py [hunk 2]
```diff
@@ -1,21 +1,49 @@
...
+        self.preferences = {}
+        for voter in self.election.voters.all().prefetch_related('preferences__candidate'):
+            self.preferences[voter.pk] = {}
+            for preference in voter.preferences.all():
+                self.preferences[voter.pk].update({
+                    preference.candidate.pk: preference.preference
+                })
+
+    def calculate_committee_score_from_prefetched(self, voter, committee, p_parameter, candidates_number):
```
Comment (jakubste) [inline ecs/elections/algorithms/brute_force.py]: test me

Comment (kasptom) [inline ecs/elections/algorithms/brute_force.py]: Tested on 40 candidates, 50 voters and 4 places:
* 1 - Brute force - 208.537s
* 5 - Brute force - 991.5s
* 10 - Brute force - 962.776s
* 15 - Brute force - 978.187s
* 20 - Brute force - 986.224s
* 40 - Brute force - 1055.783s
elections_brute_c40_v50_pl4_p1-80

Comment (jakubste) [inline ecs/elections/algorithms/brute_force.py]: my comment was about unittests, but it's quite a pretty gif :dancer:

**Comment by jakubste** [comment]
Okay, @kasptom: please review this and merge if it's okay. I will add a task to add unittests to BruteForce when somebody will have some free time.

**Pull request merged**